### PR TITLE
murdock: don't run static tests

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

Static Tests are already executed by Travis.
Failing Travis will already block the merge, so we don't have to run those tests in Murdock again.

This has two benefits:

 - slightly decreased CI build times
 - Murdock failures will only be caused by build errors.

The last point is my main motivation for this patch.
It often happens that Murdock failures are dismissed during review if a PR is not squashed yet.
Even if there are still build failures, they are not visible at a glance.

Travis will already fail when the PR is not yet squashed, so Murdock doesn't have to check this again and can now only warn about serious issues.

### Testing procedure

 - Murdock should not fail if there is a !fixup commit.
 - Travis should fail if there is a !fixup commit.

### Issues/PRs references

none

### CODE REVIEW

**Comment by aabadie** [comment]
> Failing Travis will already block the merge

Not really, Travis is not required to be green in the project configuration but that can be changed.

+1 for this PR.

File: .murdock [hunk 1]
```diff
@@ -339,7 +339,7 @@ static_tests() {
         exit 1
     fi
 
-    BUILDTEST_MCU_GROUP=static-tests ./dist/tools/ci/build_and_test.sh
+    # Travis will already run static tests
```
Comment (miri64) [inline .murdock]: Why not remove the whole function `static_tests()` altogether?

Comment (miri64) [reply .murdock]: BTW... I think this skips the rebase step included in the static tests.

**Comment by kaspar030** [comment]
> It often happens that Murdock failures are dismissed during review if a PR is not squashed yet.
> Even if there are still build failures, they are not visible at a glance.

How do you mean? Murdock build failures are visible in the last PR build result. Or do you mean possible warnings in the static test?

File: .murdock (hunk 1 above)
Comment (benpicco) [reply .murdock]: It seems like the function also takes care of cloning the repo.
TBH I'm not very familiar with the inner workings of Murdock, so my bet was to invoke Cunningham's Law on this one.

**Comment by kaspar030** [comment]
This also skips the static tests in nightlies.

I don't like it, I prefer a single source of authority.

**Comment by benpicco** [comment]
> How do you mean? Murdock build failures are visible in the last PR build result. Or do you mean possible warnings in the static test?

If the PR contains !fixup commits, Murdock failure is to be expected, so often a Murdock failure is ignored even though it was caused by a build failure.
That just means the PR needs to go another round unnecessarily, because the failure could have been caught much earlier.

**Comment by miri64** [comment]
> If the PR contains !fixup commits, Murdock failure is to be expected, so often a Murdock failure is ignored even though it was caused by a build failure.
> That just means the PR needs to go another round unnecessarily, because the failure could have been caught much earlier.

Isn't that basically more of a graphic design issue? We could just make the compile errors more prominent (and not put them to the very end of `output.html`.

**Comment by benpicco** [comment]
> Isn't that basically more of a graphic design issue?

Sure!

> We could just make the compile errors more prominent (and not put them to the very end of `output.html`).

I'm talking about this interface:

Download

If you expect the Murdock failure to be just because of !fixups, you don't click the link to get more details.

Another option would be to make Murdock reply to the PR directly.
That would also help new users who don't know about Murdock yet, but I don't know how to implement that.

**Comment by kaspar030** [comment]
> If you expect the Murdock failure to be just because of !fixups, you don't click the link to get more details.

I've been thinking of adding more fine grained PR statuses. I was mostly concerned with warnings, which are currently easily ignored.

This needs some setup, but we could add anything here.

**Comment by miri64** [comment]
> If you expect the Murdock failure to be just because of !fixups, you don't click the link to get more details.

Ok, but if you don't even look at the results and expect everything to be perfect then its a PEBCAK.

> Another option would be to make Murdock reply to the PR directly.
> That would also help new users who don't know about Murdock yet, but I don't know how to implement that.

It kind of does already in the status. If the build fails it says "the build failed", if only the static tests fail it says "static tests failed" IIRC.

File: .murdock (hunk 1 above)
Comment (kaspar030) [reply .murdock]: > It seems like the function also takes care of cloning the repo.

Not entirely. The static tests script (```./dist/tools/ci/build_and_test.sh```) tests if the current branch can be rebased on master. It does so by rebaseing on upstream/master.
"static_tests()" just ensures that there's an upstream and it is correctly fetched.

**Pull request closed without merging**